lib/bolt/transport/orch.rb

```diff
@@ -113,7 +116,11 @@ def batch_upload(targets, source, destination, _options = {}, &callback)
       end
 
       def batches(targets)
-        targets.group_by { |target| target.options[:orch_task_environment] }.values
+        targets.group_by do |target|
+          [target.options[:orch_task_environment],
+           target.options[:"service-url"],
+           target.options[:"token-file"]].join(' * ')
```

Comment:
I don't think you need to join, group_by can handle arrays as keys.

Reply:
Is the idea to squash out nil values?

Reply:
I guess if it's worth keeping, I'd like a comment why it's useful over grouping by the tuple of those values.

Reply:
at one point I was logging it. I can remove it now